### PR TITLE
ER-10: Equipment Addition API

### PR DESCRIPTION
**Pull request by YiJiePV**
```
POST/ADD TOOL
URL => /api/inventory/add
Request Body:
{
     "schoolId": "901234123"
     "serialId": "mi15Op3a",
     "typeId": x,
     "modelId": x,
     "maintenanceStatus": "Ready" OR "Under Repair",
     "reservationStatus": "Available" OR "In Use",
     "usageCondition": "Used" OR "New",
     "purchaseCost": 1102.23,
     "purchaseDate": "2016-12-08"
 }
```
I implemented the inventory equipment add API. I also created a test suite in Postman that test various different test cases against this API.
If you want to test this API with the test suite, start up the server with `npm start`, then locate and run each test request in the `ETC_API/Inventory/Add_Tool_Tests` directory. You will have to change the `typeId` and `modelId` parameters, since the items associated with them have been deleted from the database ('cause they were bogus stand-ins for testing purposes 😜). You can change the values however you wish; feel free to try all kinds of things to break this API 😉.
Because of limited time, there's probably stuff I've missed even with all my test cases, so if you find any, please let me know! Thanks!

### CODE REVIEW

File: src_code/src/routes/InventoryRoutes.js
Comment (tranleduyan) [inline src_code/src/routes/InventoryRoutes.js]: I will recommend the endpoint to be something like this: 
=> /api/inventory/equipments 
then a post request. 
This helps us make things consistent with other endpoints such as Create model is something like 
=> /api/inventory/models

File: src_code/src/routes/InventoryRoutes.js
Comment (tranleduyan) [inline src_code/src/routes/InventoryRoutes.js]: We are expected to use the term "Equipment" instead of "Tool". Please adjust this in other code, to make this match the term we are expected to use.

File: src_code/src/utils/services/helper_functions/db_helper_functions/type/TypeDBHelperFunctions.js
Comment (tranleduyan) [inline src_code/src/utils/services/helper_functions/db_helper_functions/type/TypeDBHelperFunctions.js]: We already have the function GetTypeInfoById which include the responseObject that you want to return in this GetTypeById function here. I wonder do we really need this function?

File: src_code/src/controllers/services/Inventory/AddTool.js
Comment (tranleduyan) [inline src_code/src/controllers/services/Inventory/AddTool.js]: It will be nice if we have a little description of the function here.

File: src_code/src/controllers/services/Inventory/AddTool.js
Comment (tranleduyan) [inline src_code/src/controllers/services/Inventory/AddTool.js]: See other comments about the term we are using.

File: src_code/src/controllers/services/Inventory/AddTool.js
Comment (tranleduyan) [inline src_code/src/controllers/services/Inventory/AddTool.js]: fix this comments.

File: src_code/src/controllers/services/Inventory/AddTool.js
Comment (tranleduyan) [inline src_code/src/controllers/services/Inventory/AddTool.js]: I don't really think we need to let the client knows about what we expected. This helps us secured our code (hide the information, prevent from bad people having enough information and try to making request to our server then manipulating our database). We also render these errors response to the front-end (where user will actually see, so it will be non-sense if they see something like "Expected...", they probably don't know what to do. Same for other expected message.

File: src_code/src/utils/interfaces/IDBHelperFunctions.js
Comment (tranleduyan) [inline src_code/src/utils/interfaces/IDBHelperFunctions.js]: It'll be nice if we can keep the name consistent with Equipment instead of Equip.

File: src_code/express_app/app.js
Comment (tranleduyan) [inline src_code/express_app/app.js]: Don't push anything changes on the port, on your dev side during QA. We can use any port we want as long as it works good for us. But we still keep 5000 as constant port number.

File: src_code/src/controllers/interfaces/IInventory.js
Comment (tranleduyan) [inline src_code/src/controllers/interfaces/IInventory.js]: I will suggest this to something like "EquipmentAdditionService" instead of "addToolService"

File: src_code/src/controllers/interfaces/IInventory.js
Comment (tranleduyan) [inline src_code/src/controllers/interfaces/IInventory.js]: See other comments for information about our expected term to use.

File: src_code/src/controllers/services/Inventory/AddTool.js
Comment (tranleduyan) [inline src_code/src/controllers/services/Inventory/AddTool.js:1]: This file can be rename to EquipmentAddition.js

**Review by tranleduyan — CHANGES_REQUESTED** [review]
There are some that we need to make changes on to ensure consistence.

File: src_code/express_app/app.js
Comment (mohammedking11-spu) [reply src_code/express_app/app.js]: I agree with Le that 5000 should be the constant port number.

File: src_code/src/controllers/interfaces/IInventory.js
Comment (mohammedking11-spu) [reply src_code/src/controllers/interfaces/IInventory.js]: Le's suggested name seems more reasonable. Might be subjective but that is what I think.

File: src_code/src/utils/interfaces/IDBHelperFunctions.js
Comment (mohammedking11-spu) [reply src_code/src/utils/interfaces/IDBHelperFunctions.js]: I agree with Le, consistency is key.

File: src_code/src/utils/services/helper_functions/db_helper_functions/equipment/EquipmentDBHelperFunctions.js
Comment (mohammedking11-spu) [inline src_code/src/utils/services/helper_functions/db_helper_functions/equipment/EquipmentDBHelperFunctions.js:1]: Great comments and abides with code style convention.

File: src_code/src/utils/services/helper_functions/db_helper_functions/model/ModelDBHelperFunctions.js
Comment (mohammedking11-spu) [inline src_code/src/utils/services/helper_functions/db_helper_functions/model/ModelDBHelperFunctions.js:1]: Easy to follow, good job!

File: src_code/src/utils/services/helper_functions/db_helper_functions/type/TypeDBHelperFunctions.js
Comment (mohammedking11-spu) [reply src_code/src/utils/services/helper_functions/db_helper_functions/type/TypeDBHelperFunctions.js]: I was wondering the same thing as Le. What other purpose does this function serves that GetTypeInfoById doesn't serve?

File: src_code/src/utils/services/helper_functions/general_helper_functions/GeneralHelperFunctions.js
Comment (mohammedking11-spu) [inline src_code/src/utils/services/helper_functions/general_helper_functions/GeneralHelperFunctions.js:1]: Looks good for me.  Very nicely written.

**Review by mohammedking11-spu — CHANGES_REQUESTED** [review]
It looks good so far, there are few things that needs to be resolved before this PR gets my approval. Please take a look at both my comments and Le's comments. Other than that, the style looks great and it was well documented.

File: src_code/src/routes/InventoryRoutes.js
Comment (YiJiePV) [reply src_code/src/routes/InventoryRoutes.js]: Ah, I see. I think that's a much better idea. I'll make that change, and perhaps make a note of that in Confluence

File: src_code/src/routes/InventoryRoutes.js
Comment (YiJiePV) [reply src_code/src/routes/InventoryRoutes.js]: Hmm, I see. I've been getting those two terms confused since our product is titled "ECS Tool Checkout", but we usually use "equipment" when discussing the app. So I guess we stick with "equipment" for dev purposes, and "tool" when communicating with the client or something?

File: src_code/src/utils/services/helper_functions/db_helper_functions/type/TypeDBHelperFunctions.js
Comment (YiJiePV) [reply src_code/src/utils/services/helper_functions/db_helper_functions/type/TypeDBHelperFunctions.js]: I think the only difference is that this `GetType` doesn't search for any models and equipment with the type ID. I thought that this version is more time-efficient for the task I'm using it for, but I can remove it for space's sake.

File: src_code/src/controllers/services/Inventory/AddTool.js
Comment (YiJiePV) [reply src_code/src/controllers/services/Inventory/AddTool.js]: Whoops, forgot to add that! Thanks for letting me know 👍

File: src_code/src/controllers/services/Inventory/AddTool.js
Comment (YiJiePV) [reply src_code/src/controllers/services/Inventory/AddTool.js]: Whoops, I thought I already modified this comment, but I guess I've forgotten a few parts. Thanks!

File: src_code/src/controllers/services/Inventory/AddTool.js
Comment (YiJiePV) [reply src_code/src/controllers/services/Inventory/AddTool.js]: Oh, is there a reason that method sends the error messages to the user? I was under the impression that these messages will be seen by only the frontend devs. (I already took a look at the code for it BTW)

File: src_code/src/utils/interfaces/IDBHelperFunctions.js
Comment (YiJiePV) [reply src_code/src/utils/interfaces/IDBHelperFunctions.js]: Got it! 👍

File: src_code/express_app/app.js
Comment (YiJiePV) [reply src_code/express_app/app.js]: Oh, okay. So 5001 is for our own individual development purposes and 5000 is reserved for the deployed product?

File: src_code/src/controllers/interfaces/IInventory.js
Comment (YiJiePV) [reply src_code/src/controllers/interfaces/IInventory.js]: Yeah, I'll definitely change that when I've changed the file name itself, like Le pointed out below.

File: src_code/src/utils/services/helper_functions/general_helper_functions/GeneralHelperFunctions.js
Comment (YiJiePV) [reply src_code/src/utils/services/helper_functions/general_helper_functions/GeneralHelperFunctions.js:1]: Thanks!

File: src_code/express_app/app.js
Comment (mohammedking11-spu) [reply src_code/express_app/app.js]: Yup.

File: src_code/src/controllers/interfaces/IInventory.js
Comment (tranleduyan) [inline src_code/src/controllers/interfaces/IInventory.js]: This can be EquipmentAdditionService

File: src_code/src/controllers/interfaces/IInventory.js
Comment (tranleduyan) [inline src_code/src/controllers/interfaces/IInventory.js]: After making changes to the above initialize modules, you can export it here as:
EquipmentAddition: EquipmentAdditionService.EquipmentAddition

File: src_code/src/controllers/services/Inventory/EquipmentAddition.js
Comment (tranleduyan) [inline src_code/src/controllers/services/Inventory/EquipmentAddition.js:151]: Very nice break down of code!

File: src_code/src/controllers/services/Inventory/EquipmentAddition.js
Comment (tranleduyan) [inline src_code/src/controllers/services/Inventory/EquipmentAddition.js]: You can just let it be EquipmentAddition if you want :) since they are both the same.

File: src_code/src/utils/services/helper_functions/db_helper_functions/equipment/EquipmentDBHelperFunctions.js
Comment (tranleduyan) [inline src_code/src/utils/services/helper_functions/db_helper_functions/equipment/EquipmentDBHelperFunctions.js:7]: Maybe the const tool can be const equipment?

**Review by tranleduyan — CHANGES_REQUESTED** [review]
Thank you so much for addressing the comments, Karena! It is looking good now, but there are just like a few name can be changed that I think you would have missed! After those are addressed, I think it should be good to go!

**Review by mohammedking11-spu — APPROVED** [review]
For the sake of time; This PR looks great and written nicely, I will approve it on my end but I entrust you to review Le's comments so that as soon as they are resolved, this PR will be approved faster since it took too long. Other than that, good comments and very nice structure.

File: src_code/src/controllers/services/Inventory/EquipmentAddition.js
Comment (YiJiePV) [reply src_code/src/controllers/services/Inventory/EquipmentAddition.js:151]: Thanks!

**Review by tranleduyan — APPROVED** [review]
It looks good now!

**Pull request merged**
